pkg/cmd/grafana-server/server.go

```diff
@@ -55,6 +57,12 @@ func (g *GrafanaServerImpl) Start() {
 	g.writePIDFile()
 
 	initSql()
+	err := dsSettings.Init(filepath.Join(setting.HomePath, "conf/datasources.yaml"))
```

Comment:
think I would prefer this as a service and initialized like: 

```go
setupFromFilesSrv := setupFromFilesSrv.New(setting.HomePath, "conf/datasources.yaml")
setupFromFilesSrv.childRoutines.Go(func() error { return engine.Run(g.context) })
```

Thinking is that we might want Grafana to detect changes to these files and sync without restarting Grafana. Aslo naming the service something like setupFromFilesSrv (or a better name) makes it more open to be used for Users etc.

Reply:
I don't want to include auto reloading in the scope of this PR. But once this PR is merged it should be easy to start working on that.

Reply:
I am fine with that, but still think it should be a service  you create, and not a static package Init function as we are trying to move away from that

Reply:
You right.

The example code you posted starts the process as a go routine. Wouldn't you want this to be blocking? If the setup fails, shouldn't grafana start up fail? I can see arguments for both answers. But I lean towards start up failing cause if you configure data source this way you really expected those to exist.

Reply:
Yes, your right about that. Should block startup